bentoml/_internal/io_descriptors/pandas.py

```diff
@@ -483,11 +487,117 @@ def predict(inputs: pd.DataFrame) -> pd.DataFrame: ...
 
         return inst
 
-    async def from_proto(self, request) -> t.Any:
-        pass
+    async def from_proto(self, request: pb.Request) -> ext.PdDataFrame:
+        """
+        Process incoming protobuf request and convert it to ``pandas.DataFrame``
 
-    async def to_proto(self, obj) -> t.Any:
-        pass
+        Args:
+            request: Incoming RPC request message.
+            context: grpc.ServicerContext
+
+        Returns:
+            a ``pandas.DataFrame`` object. This can then be used
+             inside users defined logics.
+        """
+        contents = {}
+        if request.HasField("dataframe"):
+            if self._orient != "columns":
+                raise BadInput(
+                    'Currently using `dataframe` field only supports columns orient. Try using `orient = "columns"` in service definition or `raw_bytes_contents` field instead.'
+                )
+
+            column_names = request.dataframe.column_names
+            column_values = request.dataframe.columns
+
+            # try to use provided `self._columns` if `self._apply_column_names` is true
+            if self._apply_column_names:
+                if self._columns is None:
+                    logger.warning(
+                        "`columns` is None or undefined, while `apply_column_names`=True"
+                    )
+                else:
+                    column_names = self._columns
+
+            if len(column_names) != len(column_values):
+                raise UnprocessableEntity(
+                    "The length `column_names` provided doesn't match the length of `column` series contents."
+                )
+
+            # contents(column orient) { column_name : {index : columns.series._value}}
+            for index, content in enumerate(column_values):
+                fieldpb = [
```

Comment:
Do we have to iterate through every single contents? is there a way to do this in batch?

Reply:
I can't think of any other way. Maybe we can try using `MessageToDict` to change the field to a dictionary?

Reply:
That's the same either way right?

Reply:
Would require iterating through the dict.

Reply:
which is essentially the same. You still have to iterate through the whole item on way or another. My concern is that since we aren't doing this in-place, copying large dataframe is very expensive.

Reply:
Is there anything we can do with pandas in-place?

Reply:
https://pandas.pydata.org/docs/reference/api/pandas.read_json.html
Assuming the user uses to_json, we can use json string to serialize similar to HTTP. However since the client can be in languages other than python, I don't know if we want to do this.